### PR TITLE
Fix proof generation for subsequential polls

### PR DESCRIPTION
**Pull request by corydickson**
Instead of allowing concurrent polls where users must signup for each poll, this PR aims to allow on chain proof verification for a multiple poll instances with the same set of users.

Testing run `cli/testMultiplePoll.sh`

### CODE REVIEW

File: cli/testMultiplePoll.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+node build/index.js publish \
+    -p macipk.3e7bb2d7f0a1b7e980f1b6f363d1e3b7a12b9ae354c2cd60a9cfa9fd12917391 \
+    -sk macisk.fd7aa614ec4a82716ffc219c24fd7e7b52a2b63b5afb17e81c22fe21515539c \
+    -x 0xf204a4Ef082f5c04bB89F7D5E6568B796096735a \
+    -i 1 -v 0 -w 7 -n 1 -o 1
+
+node build/index.js timeTravel -s 140 && \
+node build/index.js mergeMessages -x 0xf204a4Ef082f5c04bB89F7D5E6568B796096735a -o 1 && \
+node build/index.js mergeSignups -x 0xf204a4Ef082f5c04bB89F7D5E6568B796096735a -o 1 && \
```
Comment (corydickson) [inline cli/testMultiplePoll.sh:62]: Without changing how the sbCommitments are updated, this command is still necessary even though the StateAq subtrees are merged.

**Comment by weijiekoh** [comment]
Hi! Does this include https://github.com/appliedzkp/maci/pull/311 or should that be merged separately?

**Comment by weijiekoh** [comment]
`PollFactory` compiles to 25Kb which exceeds the mainnet limit

**Review by weijiekoh — DISMISSED** [review]

**Comment by weijiekoh** [comment]
Suggestions to reduce the size of Poll.sol:

- Comment out `batchEnqueueMessage` which isn't used - size is now 24.57 Kb
- Have a separate ResultVerifier contract that implements `verifySpentVoiceCredits`, `verifyPerVOSpentVoiceCredits`, `verifyTallyResult`, and `computeMerkleRootFromPath`, instead of implementing these functions in Poll.sol.

**Review by weijiekoh — CHANGES_REQUESTED** [review]
Looks good and the test script works on my end too. We just need to reduce the size of Poll.sol - see the previous comment on how we can do this.

Edit: let's do the size reduction in a different PR.

**Pull request merged**